src/compiler/types.rs

```diff
@@ -8,13 +8,16 @@ impl Path {
     }
 }
 
-#[derive(Copy, Clone, Debug)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
```

Comment:
I saw that you were recommending deriving `Eq`/`PartialEq` to be able to use `==` to compare enum variants